algorithms/refinement/outlier_detection/outlier_base.py

```diff
@@ -438,14 +508,31 @@ def from_parameters_and_colnames(params, colnames):
             k: v for k, v in algo_params.__dict__.items() if not k.startswith("_")
         }
 
+        # Special case where we ignore experiment and blocks and just look
+        # for outliers on each image independently
+        if params.outlier.separate_images:
+            if params.outlier.separate_blocks:
+                logger.info("Resetting separate_blocks=False")
+                params.outlier.separate_blocks = False
+            if params.outlier.separate_experiments:
+                logger.info("Resetting separate_experiments=False")
+                params.outlier.separate_experiments = False
+
         if not params.outlier.separate_blocks:
             params.outlier.block_width = None
+
+        if params.outlier.nproc is libtbx.Auto:
+            params.outlier.nproc = available_cores()
```

Comment:
Hi, I'm excited by this pull request.  Does this line mean that dials.stills_process will use available_cores() when doing outlier rejection?  This doesn't work in MPI multiprocessing, where each process would independently call available_cores and then use too many.  But I haven't tested it :)

Reply:
Hi @phyy-nx, the PR for _that_ change specifically is https://github.com/dials/dials/pull/2031, where you are tagged for review :) (I appreciate you are in beam time mode so won't have had chance to look).

In #2031, `dials.refine` is changed to automatically select `nproc` for outlier rejection, but that default is restricted to the command line program. Use of the `Refiner` as a library has unchanged defaults. In fact, when `nproc=1` the outlier rejection is done in the main process. I found that was necessary to get identical results for those outlier algorithms that use random sampling (like `mcd`), plus it avoids overhead of setting up a subprocess.

Reply:
Thanks @dagewa for clarifying!  Not surprising you considered this.  Indeed beamtimes :)